### PR TITLE
Allow enabling mesh internal identity propagation

### PR DESCRIPTION
**Pull request by smarunich**
Addressing https://github.com/tetrateio/tetrate-service-bridge-sandbox/issues/225

`cr-pull-secret` has to be refreshed periodically for `AWS`, `cr-pull-secret` PR: https://github.com/tetrateio/tetrate-service-bridge-sandbox/pull/164

More details around secret management caveats: https://github.com/tetrateio/xcp/issues/2001

### CODE REVIEW

File: modules/tsb/cp/manifests/tsb/controlplane-values.yaml.tmpl [hunk 1]
```diff
@@ -34,7 +38,9 @@ spec:
     xcp:
       centralAuthMode: JWT
       configProtection: {}
-      enableHttpMeshInternalIdentityPropagation: false
+      %{ if id_propagation_enabled }
+      enableHttpMeshInternalIdentityPropagation: true
+      %{ endif }
```
Comment (nacx) [inline modules/tsb/cp/manifests/tsb/controlplane-values.yaml.tmpl]: [nit], less verbose:
```suggestion
      enableHttpMeshInternalIdentityPropagation: ${id_propagation_enabled}
```

File: tsb/cp/variables.tf [hunk 2]
```diff
@@ -136,3 +136,7 @@ variable "cert-manager_enabled" {
 variable "ratelimit_enabled" {
   default = true
 }
+
+variable "id_propagation_enabled" {
```
Comment (nacx) [inline tsb/cp/variables.tf]: [nit] I'd suggest you rename this to `identity_propagation_enabled `, as `id` looks confusing to me. What ID? Could it refer to the request ID across hops? `identity_propagation_enabled` reads better and is more understandable.

Comment (smarunich) [reply tsb/cp/variables.tf]: ack, fixed

File: modules/tsb/cp/manifests/tsb/controlplane-values.yaml.tmpl (hunk 1 above)
Comment (smarunich) [reply modules/tsb/cp/manifests/tsb/controlplane-values.yaml.tmpl]: makes sense, overlooked at the start...

**Review by nacx — APPROVED** [review]

**Pull request merged**